packages/stellar_client/lib/src/client.dart

```diff
@@ -0,0 +1,440 @@
+      return true;
+    } catch (error) {
+      print("Failed to create account $error");
+      return false;
+    }
+  }
+
+  // take asset code code && issuer (optional)
+  Future<bool> fundTrustLine(String asset_code) async {
+    final asset = _currencies.currencies[asset_code];
+    if (asset == null) {
+      throw Exception("Asset $asset_code not supported");
+    }
+
+    try {
+      final response = await http.post(
+        Uri.parse(
+            '${_serviceUrls[_network.toString()]}/activation_service/fund_trustline'),
+        headers: {'Content-Type': 'application/json'},
+        body: jsonEncode({
+          'asset': '${asset.assetCode}:${asset.issuer}',
+          'address': accountId
+        }),
+      );
+
+      final data = jsonDecode(response.body);
+
+      String trustlineTransaction = data['addtrustline_transaction'];
+      XdrTransactionEnvelope xdrTxEnvelope =
+          XdrTransactionEnvelope.fromEnvelopeXdrString(trustlineTransaction);
+
+      Transaction transaction =
+          Transaction.fromV1EnvelopeXdr(xdrTxEnvelope.v1!);
+
+      transaction.sign(_keyPair, _stellarNetwork);
+
+      SubmitTransactionResponse response2 =
+          await _sdk.submitTransaction(transaction);
+
+      if (response2.success) {
+        print("Trustline for $asset_code added successfully.");
+        return true;
+      } else {
+        print("Failed to add trustline for $asset_code.");
+        return false;
+      }
+    } catch (error) {
+      throw Exception("Error funding trustline : $error");
+    }
+  }
+
+  Future<TransactionData?> fetchFundDetails({required String assetCode}) async {
+    try {
+      final response = await http.get(
+        Uri.parse(
+            '${_serviceUrls[_network.toString()]}/transactionfunding_service/conditions'),
+        headers: {'Content-Type': 'application/json'},
+      );
+
+      final body = jsonDecode(response.body);
+      Map<String, dynamic>? details;
+      for (var item in body) {
+        if (item['asset'].startsWith(assetCode)) {
+          details = item;
+          break;
+        }
+      }
+
+      if (details != null) {
+        TransactionData transactionData = TransactionData.fromJson(details);
+        print(transactionData);
+        return transactionData;
+      } else {
+        print('Failed to details for asset: ');
+      }
+
+      return null;
+    } catch (error) {
+      throw Exception("Failed to get fund details, $error");
+    }
+  }
+
+  Future<Operation?> makeFundPaumentOperation(
+      {required String assetCode, required String issuer}) async {
+    TransactionData? transactionData =
+        await fetchFundDetails(assetCode: assetCode);
+    Asset asset = AssetTypeCreditAlphaNum4(assetCode, issuer);
+
+    return PaymentOperationBuilder(
+            transactionData!.feeAccountId, asset, transactionData.feeFixed)
+        .build();
+  }
+
+  Future<Transaction?> buildFundedPaymentTransaction(
```

Comment:
it's doing the same as sendTransaction(transfer) method but without sending the transaction. so can we reuse it in the sendTransaction(transfer) method?